handler/src/main/java/io/netty/handler/ssl/ReferenceCountedOpenSslEngine.java

```diff
@@ -235,37 +235,39 @@ protected void deallocate() {
         } finally {
             readerLock.unlock();
         }
-        ssl = SSL.newSSL(context.ctx, !context.isClient());
-        try {
-            networkBIO = SSL.bioNewByteBuffer(ssl, context.getBioNonApplicationBufferSize());
+        synchronized (this) {
+            ssl = SSL.newSSL(context.ctx, !context.isClient());
```

Comment:
Actually, I don't understand.  `ssl` is intentionally non-final in order to destroy it during shutdown.  Are you saying you want a final _local_ variable?